jupyter_book/directive/toc.py

```diff
@@ -22,29 +29,154 @@ def run(self):
+    """
+    for key, val in globaltoc.items():
+        if key == "file":
+            if val == parent_file:
+                del globaltoc["file"]
+                return globaltoc
+        if key == "sections":
+            for item in val:
+                if "file" in item and item["file"] == parent_file:
+                    del item["file"]
+                    filtered_toc = item
+                    break
+                elif "sections" in item:
+                    filtered_toc = process_toc_dict(item, parent_file, filtered_toc)
+            if filtered_toc:
+                return filtered_toc
+    return
+
+
+def has_toc_yaml(
+    self, subnode: nodes.Element, tocdict: Dict[str, nodes.Element], depth: int
+) -> None:
+    """constructs toc nodes from globaltoc dict
+    :param subnode: node to which toc constructed here is appended to
+    :param tocdict: dictionary of toc entries
+    :param depth: current toclevel depth
+    """
+    depth += 1
+    for key, val in tocdict.items():
+        if key in ["file", "url"]:
+            internal = False
+            if "title" in tocdict:
+                title = tocdict["title"]
+            else:
+                if val not in self.env.titles:
+                    continue
+                title = clean_astext(self.env.titles[val])
+            if "url" in tocdict:
+                if "http" in tocdict["url"]:
+                    internal = False
+                else:
+                    continue
+            else:
+                val = "%" + val
+                internal = True
+            reference = nodes.reference(
+                "",
+                "",
+                internal=internal,
+                refuri=val,
+                anchorname="",
+                *[nodes.Text(title)],
+            )
+            para = addnodes.compact_paragraph("", "", reference)
+            item = nodes.list_item("", para)
+            item["classes"].append("tableofcontents-l%d" % (depth))
+            subnode.append(item)
+        if key in ["sections"]:
+            sectionlist = nodes.bullet_list().deepcopy()
+            sectionheader = None
+            for item in val:
+                if "part" in item:
+                    sectionheader = handle_toc_header(item["part"])
+                    sectionlist.append(sectionheader)
+                    del item["part"]
+                    has_toc_yaml(self, sectionlist, item, depth)
+                else:
+                    has_toc_yaml(self, sectionlist, item, depth)
+            subnode.append(sectionlist)
+
+
+def handle_toc_header(val: str) -> nodes.Element:
+    """Constructs node for the headers in globaltoc
+    :param val: value of the node
+    """
+    para = addnodes.compact_paragraph("", "", nodes.Text(val))
+    item = nodes.list_item("", para)
+    item["classes"].append("fs-1-2")
+    return item
+
+
 class SwapTableOfContents(SphinxTransform):
     default_priority = 700
```

Comment:
Can you add a comment on why 700 was chosen. Is there anything specific it needs to come before/after?

Reply:
No particular reason. I think I was trying something out but the priority did not matter, should probably remove it?

Reply:
I think its required to have a priority, you see the range categories here https://www.sphinx-doc.org/en/master/extdev/appapi.html?highlight=config-inited#sphinx.application.Sphinx.add_transform

Reply:
@chrisjsewell can you help with what priority to choose 😬. Ideally would want it to be the lowest possible one for the given functionality, right?

Reply:
thanks @chrisjsewell that is a helpful link. @AakashGfude perhaps you should set this in the range of `900-999` unless you don't need `referencing` resolution given `LaTeX` probably doesn't need that.